src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -57,6 +57,76 @@ function migrateIndexPattern(doc) {
   doc.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(searchSource);
 }
 
+// [TSVB] Migrate percentile-rank aggregation (value -> values)
+const migratePercentileRankAggregation = doc => {
+  const visStateJSON = get(doc, 'attributes.visState');
+  let visState;
+
+  if (visStateJSON) {
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+    if (visState && visState.type === 'metrics') {
+      const series = get(visState, 'params.series') || [];
+
+      series.forEach(part => {
+        (part.metrics || []).forEach(metric => {
+          if (metric.type === 'percentile_rank' && has(metric, 'value')) {
+            metric.values = [metric.value];
+
+            delete metric.value;
+          }
+        });
+      });
+      return {
+        ...doc,
+        attributes: {
+          ...doc.attributes,
+          visState: JSON.stringify(visState),
+        },
+      };
+    }
+  }
+  return doc;
+};
+
+// Migrate date histogram aggregation (remove customInterval)
+const migrateDateHistogramAggregation = doc => {
+  const visStateJSON = get(doc, 'attributes.visState');
+  let visState;
+
+  if (visStateJSON) {
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+
+    if (visState && visState.aggs) {
+      visState.aggs.forEach(agg => {
+        if (agg.type === 'date_histogram' && agg.params && agg.params.interval === 'custom') {
+          agg.params.interval = agg.params.customInterval;
+          delete agg.params.customInterval;
+        }
+
+        if (get(agg, 'params.customBucket.type', null) === 'date_histogram'
+          && agg.params.customBucket.params
+          && agg.params.customBucket.params.interval === 'custom'
+        ) {
+          agg.params.customBucket.params.interval = agg.params.customBucket.params.customInterval;
+          delete agg.params.customBucket.params.customInterval;
+        }
+      });
+      doc.attributes.visState = JSON.stringify(visState);
```

Comment:
Migrations shouldn't mutate their arguments. So, this should be:

```js
return {
  ...doc,
  attributes: {
    ...doc.attributes,
    visState: JSON.stringify(visState),
  }
};
```

In fact, the previous migration in this file does this correctly. It might be worth creating a helper function to do this.